scripts/generateIcons.js

```diff
@@ -0,0 +1,197 @@
+            throw new Error(`${svgName}: Invalid ${attr} color: ${$.html(el)}`);
+          }
+        });
+      });
+
+      const iconName = `${pascalCase(svgName)}Icon`;
+      const svgComponentName = `${iconName}${
+        variantName ? pascalCase(variantName) : ''
+      }Svg`;
+      const svgComponent = await svgr(optimisedSvg, svgrConfig, {
+        componentName: svgComponentName,
+      });
+
+      // Create icon directory if it's missing
+      const iconDir = path.join(iconComponentsDir, iconName);
+      await fs.mkdirp(iconDir);
+
+      // Write SVG React component
+      await fs.writeFile(
+        path.join(iconDir, `${svgComponentName}.tsx`),
+        svgComponent,
+        { encoding: 'utf-8' },
+      );
+
+      // Bail out early if we're processing an icon variant (e.g. bookmark-active.svg)
+      // All subsequent steps should only happen once per icon component.
+      if (variantName) {
+        return;
+      }
+
+      const templateFileIfMissing = async (relativePath, contents) => {
+        const filePath = path.join(iconDir, relativePath);
+        if (!(await fs.exists(filePath))) {
+          await fs.writeFile(filePath, `${contents}\n`, {
+            encoding: 'utf-8',
+          });
+        }
+      };
+
+      // Create icon wrapper component, if it doesn't already exist
+      await templateFileIfMissing(
+        `${iconName}.tsx`,
+        dedent`
+          import React from 'react';
+          import { Box } from '../../Box/Box';
+          import useIcon, { UseIconProps } from '../../../hooks/useIcon';
+          import { ${svgComponentName} } from './${svgComponentName}';
+
+          export type ${iconName}Props = UseIconProps;
+
+          export const ${iconName} = (props: UseIconProps) => {
+            const iconProps = useIcon(props);
+
+            return <Box component={${svgComponentName}} {...iconProps} />;
+          };
+        `,
+      );
+
+      // Create documentation, if it doesn't already exist
+      await templateFileIfMissing(
+        `${iconName}.docs.tsx`,
+        dedent`
+          import { ComponentDocs } from '../../../../site/src/types';
+          import examplesForIcon from '../../private/examplesForIcon';
+          import { ${iconName} } from './${iconName}';
+
+          const docs: ComponentDocs = {
+            migrationGuide: true,
+            examples: examplesForIcon(${iconName}),
+          };
+
+          export default docs;
+        `,
+      );
+
+      // Create migration guide, if it doesn't already exist
+      await templateFileIfMissing(
+        `${iconName}.migration.md`,
+        dedent`
+          # ${iconName} Migration Guide
+
+          Please refer to the [Icon Migration Guide.](../Icon.migration.md)
+        `,
+      );
+    }),
+  );
+
+  // Create icons/index.ts
+  const files = await fs.readdir(iconComponentsDir);
+  const iconComponents = files.filter(
+    file => file.includes('Icon') && !file.includes('.'),
```

Comment:
`file.endsWith('Icon.tsx')`?

Reply:
Updated the code to make it clearer what's going on here.